src/NServiceBus.Persistence.DynamoDB/SynchronizedStorage/StorageSession.cs

```diff
@@ -74,9 +74,14 @@ public async Task Commit(CancellationToken cancellationToken = default)
 
         public void Dispose()
         {
+            if (disposed)
```

Comment:
@danielmarbach I went with this approach instead of clearing the collection. I felt this is the safer approach as we don't need to do very strict locking checks in cases where dispose might be called multiple times in parallel. In such cases, this approach ensures that concurrent attempts might still cause multiple calls to the cleanup actions (which are safe to call multiple times due to concurrency checks), but there won't be any exceptions thrown due to potential modifications of a collection while a concurrent thread might still enumerate it.

Reply:
I'm fine with that

Reply:
it should never happen unless somebody is really messing with the session. In such a case I'd rather use more resources over failing with an unnecessary exception though.